### PR TITLE
in-game update checker

### PR DESCRIPTION
**Pull request by alexban011**
Is this something we want to do?

<strong>I made it check if the game was installed from (by) google play and it only displays the button if its not.</strong>

Now that 4.1.12 has been released and I still had a built jar from 4.1.11 (and on my phone) I managed to test and it successfully detected the update.

The button only appears when there is a new update
NOTE: the update button is now the same color as the civilopedia button 

image

### CODE REVIEW

**Comment by OptimizedForDensity** [comment]
I like the idea, but there's an issue with Google Play updates not being rolled out everywhere at the same time. These users would see the popup and try to update the game through the GP app but see no update is available and think that the update checker is bugged.

**Comment by alexban011** [comment]
> Google Play updates not being rolled out everywhere

yeah thats why I'm thinking to either disable this on android or check for signature (if thats even possible) and only display the update checker if the user installed the apk from github

**Comment by alexban011** [comment]
ohh the signature might be secret. Well desktop only it is then

**Comment by alexban011** [comment]
No update available:
image

New update available:
image

**Comment by alexban011** [comment]
I don't understand the new ConfirmPopup...

Why doesn't this code open a confirmPopup ?
```Kotlin
ConfirmPopup(this, "Open browser to download the new update?", "Open browser") {
                    Gdx.net.openURI("https://github.com/yairm210/Unciv/releases/latest")
                }
```

**Comment by touhidurrr** [comment]
For android since Github version cannot best installed one top of Play Store version, I recommend checking if it's the Github version before showing the button maybe?

Do we really want Play Store users to go to Github & Discord and ask why they cannot install the APK file from there? Maybe not.

If someone is using Play Store, then just let them do so.

**Comment by touhidurrr** [comment]
> If someone is using Play Store, then just let them do so.

Disclaimer: Contains Opinions (Lol)

**Comment by alexban011** [comment]
It checks if the game is installed from GP and doesn't display those buttons if so

File: android/src/com/unciv/app/PlatformSpecificHelpersAndroid.kt [hunk 1]
```diff
@@ -56,6 +55,23 @@ Sources for Info about current orientation in case need:
...
+        val validInstallers: List<String> =
+                ArrayList(listOf("com.android.vending", "com.google.android.feedback"))
+
+        // The package name of the app that has installed Unciv
+        val installer: String? =
+                activity.window.context.packageManager.getInstallerPackageName(activity.window.context.packageName)
+
+        // true if Unciv has been downloaded from Play Store
+        return installer != null && validInstallers.contains(installer)
+    }
```
Comment (alexban011) [inline android/src/com/unciv/app/PlatformSpecificHelpersAndroid.kt:73]: @touhidurrr

**Comment by touhidurrr** [comment]
> It checks if the game is installed from GP and doesn't display those buttons if so

Oh, does it work?

**Comment by alexban011** [comment]
> Oh, does it work?

this unfortunately is something that can only be tested once GP installs the game with the PR

File: core/src/com/unciv/MainMenuScreen.kt [hunk 2]
```diff
@@ -259,4 +292,31 @@ class MainMenuScreen: BaseScreen(), RecreateOnResize {
...
+            val url = "https://github.com/yairm210/Unciv/releases/latest"
+            response = URL(url).openStream()
+            val scanner = Scanner(response)
+            val responseBody = scanner.useDelimiter("\\A").next()
+
+            version = responseBody.substring(
+                responseBody.indexOf("<title>") + 7,
+                responseBody.indexOf("</title>")
+            )
+            version = version.removePrefix("Release ").removeSuffix(" · yairm210/Unciv · GitHub")
```
[the remark this answers is not in the document]
Comment (touhidurrr) [reply core/src/com/unciv/MainMenuScreen.kt:310]: +1

**Comment by alexban011** [comment]
I also changed the color to blue instead of the yellow. I think suddenly seeing a new button is noticeable enough and is more consistent

**Comment by alexban011** [comment]
I replaced it. The one I used looked kinda odd. I just kept it because I didn't want to add a png icon and couldn't find something like `↻` :))

File: core/src/com/unciv/MainMenuScreen.kt [hunk 3]
```diff
@@ -205,8 +208,35 @@ class MainMenuScreen: BaseScreen(), RecreateOnResize {
         helpButton.addTooltip(KeyCharAndCode(Input.Keys.F1), 20f)
         helpButton.setPosition(20f, 20f)
         stage.addActor(helpButton)
+
+        if (UncivGame.Current.platformSpecificHelper?.isInstalledFromGP() == false) {
+            val latestVersion = Github.getLatestVersion()
+            if (latestVersion?.tag_name!! > UncivGame.VERSION.toString()) {
```
[the remark this answers is not in the document]
Comment (alexban011) [reply core/src/com/unciv/MainMenuScreen.kt:214]: > I don't think latestVersion.tag_name looks like that.

I printed it and it only printed the version, without the build. 

I'm gonna take a closer look when I find the time. Maybe I can get the build number from github and that should work better

**Comment by alexban011** [comment]
> not sure if we even need this, though, now that we soon also have Steam auto-updating.

This is mainly for the people who don't use the Steam version

**Comment by yairm210** [comment]
Sorry, I'm against this.
I believe people should be working with package/program managers (snap, itch.io, flatpak, GP, f-droid, what have you)
Programs should be unaware of distribution methods - this isn't 2005
The real question should therefore be 'why are some people consistently downloading the game from Github', because that does sound like a problem we should be solving.

**Comment by Kuldebar** [comment]
Can we get latest releases on Google Play Store as "beta" opt-in option?  Currently the staggered updates often leave me days behind on my Chromebook/Android install.

**Pull request closed without merging**